File: core/application/application_transition_engine.py
from __future__ import annotations

from typing import Any, Dict, List


def build_application_transitions(
    states: List[Dict[str, Any]],
) -> List[Dict[str, Any]]:
    transitions: List[Dict[str, Any]] = []

    for index in range(len(states) - 1):
        src = states[index]
        dst = states[index + 1]

        transitions.append({
            "from": str(src.get("route", "")),
            "to": str(dst.get("route", "")),
            "relation": "transition",
            "order": index,
        })

    return transitions[:10000]

File: src/application/applicationTransitionEngine.ts
/**
 * Converted from Python: core/application/application_transition_engine.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "../runtime/pyCompat.js";

export function buildApplicationTransitions(states: any): any {
  var transitions: any[] = [];
  var index: any;
  for (index = 0; index < py.sub(py.len(states), 1); index++) {
    var src: any = py.at(states, index);
    var dst: any = py.at(states, py.add(index, 1));
    py.listAppend(transitions, {"from": py.toStr(py.get(src, "route", "")), "to": py.toStr(py.get(dst, "route", "")), "relation": "transition", "order": index});
  }
  return py.slice(transitions, null, 10000);
}
